src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts

```diff
@@ -29,8 +44,122 @@ export abstract class CellView extends AngularDisposable implements OnDestroy, I
+			}
+		}
+	}
+
+	protected addDecoration(range?: NotebookRange): void {
+		range = range ?? this.highlightRange;
+		if (this.output && this.output.nativeElement) {
+			this.highlightAllMatches();
+			if (range) {
+				let elements = this.getHtmlElements();
+				if (elements?.length >= range.startLineNumber) {
+					let elementContainingText = elements[range.startLineNumber - 1];
+					let markCurrent = new Mark(elementContainingText); // to highlight the current item of them all.
+
+					markCurrent.markRanges([{
+						start: range.startColumn - 1, //subtracting 1 since markdown html is 0 indexed.
+						length: range.endColumn - range.startColumn
+					}], {
+						className: findRangeSpecificClass,
+						each: function (node, range) {
+							// node is the marked DOM element
+							node.scrollIntoView({ behavior: 'smooth', block: 'center' });
+						}
+					});
+				}
+			}
+		}
+	}
+
+	protected highlightAllMatches(): void {
+		if (this.output && this.output.nativeElement) {
+			let markAllOccurances = new Mark(this.output.nativeElement); // to highlight all occurances in the element.
+			let editor = this.notebookService.findNotebookEditor(this._model.notebookUri);
+			if (editor) {
+				let findModel = (editor.notebookParams.input as NotebookInput).notebookFindModel;
+				if (findModel?.findMatches?.length > 0) {
+					let searchString = findModel.findExpression;
+					markAllOccurances.mark(searchString, {
+						className: findHighlightClass
+					});
+				}
+			}
+		}
+	}
+
+	protected removeDecoration(range?: NotebookRange): void {
+		if (this.output && this.output.nativeElement) {
+			if (range) {
+				let elements = this.getHtmlElements();
+				let elementContainingText = elements[range.startLineNumber - 1];
+				let markCurrent = new Mark(elementContainingText);
+				markCurrent.unmark({ acrossElements: true, className: findRangeSpecificClass });
+			} else {
+				let markAllOccurances = new Mark(this.output.nativeElement);
+				markAllOccurances.unmark({ acrossElements: true, className: findHighlightClass });
+				markAllOccurances.unmark({ acrossElements: true, className: findRangeSpecificClass });
+				this.highlightRange = undefined;
+			}
+		}
+	}
+
 
+	protected getHtmlElements(): any[] {
+		let hostElem = this.output?.nativeElement;
+		let children = [];
+		if (hostElem) {
+			for (let element of hostElem.children) {
+				if (element.nodeName.toLowerCase() === 'table') {
+					// add table header and table rows.
+					if (element.children.length > 0) {
+						children.push(element.children[0]);
+						if (element.children.length > 1) {
+							for (let trow of element.children[1].children) {
+								children.push(trow);
+							}
+						}
+					}
+				} else if (element.children.length > 1) {
+					children = children.concat(this.getChildren(element));
+				} else {
+					children.push(element);
+				}
+			}
+		}
+		return children;
 	}
+
+
+	protected getChildren(parent: any): any[] {
```

Comment:
These should be typed.

Also the name `getChildren` is very generic for what seems to be a specialized form of getting the children. Please at least add a doc comment, and ideally find a better name for the function

(I'd also suggest moving this out of the class - it has no reason to be here. Either put it in a generic utils type module or you can keep it in this module if you think this isn't going to be used anywhere else)

Reply:
Angular doesn't specify a type for the native html element of the ElementRef, that is why I followed what angular has which is 'any'.  And this is very specific to find and I don't see it being reused anywhere. How about getHtmlNodeChildren for getChildren?